phoenix-core/src/main/java/com/force/db/i18n/OracleUpper.java

```diff
@@ -0,0 +1,66 @@
+/* 
```

Comment:
@joshelser - my take, based on this[1], is that it's ok to include source code in an ASF project with a BSD license (as opposed to only having BSD licensed software as an external dependency). WDYT?

[1] http://apache.org/licenses/#code-developed-elsewhere-received-under-a-category-a-license-incorporated-into-apache-projects-distributed-by-apache-and-licensed-to-downstream-users-under-its-original-license

Reply:
Yup! You got it right, James. Whether we include the code in binary form or source form, for BSD, we treat them the same (propagate in LICENSE, and copyright/etc in NOTICE). If there's a license header for the file, we would also leave that, IIRC.